File: sentinel-backend/app/core/schemas/event.py
from pydantic import BaseModel, Field, model_validator
from typing import Literal, Optional
from datetime import datetime
import uuid


class EventBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    start_at: datetime = Field(..., description="ISO-8601")
    end_at: Optional[datetime] = Field(None, description="NULL для напоминаний")
    all_day: bool = False
    type: Literal["event", "reminder"] = "event"
    location: Optional[str] = Field(None, max_length=255)
    is_fixed: bool = False

    @model_validator(mode="after")
    def end_after_start(self) -> "EventBase":
        if self.end_at is not None and self.end_at <= self.start_at:
            raise ValueError("end_at must be after start_at")
        return self


class EventCreate(EventBase):
    source: Literal["user", "ai", "import"] = "user"


class EventUpdate(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None
    start_at: Optional[datetime] = None
    end_at: Optional[datetime] = None
    all_day: Optional[bool] = None
    type: Optional[Literal["event", "reminder"]] = None
    location: Optional[str] = Field(None, max_length=255)
    is_fixed: Optional[bool] = None


class Event(EventBase):
    id: uuid.UUID
    user_id: uuid.UUID
    source: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class EventList(BaseModel):
    items: list[Event]
    total: int

File: sentinel-ios/Sentinel/Packages/SentinelCore/Sources/SentinelCore/Models/Domain/Event.swift
import Foundation

public enum EventKind: String, Equatable, Sendable {
    case event
    case reminder
}

public struct Event: Equatable, Identifiable, Sendable {
    public let id: UUID
    public let userId: UUID
    public let title: String
    public let description: String?
    public let startAt: Date
    public let endAt: Date?
    public let allDay: Bool
    public let type: EventKind
    public let location: String?
    public let isFixed: Bool
    public let source: String
    public let createdAt: Date
    public let updatedAt: Date

    public init(
        id: UUID,
        userId: UUID,
        title: String,
        description: String?,
        startAt: Date,
        endAt: Date?,
        allDay: Bool,
        type: EventKind,
        location: String?,
        isFixed: Bool,
        source: String,
        createdAt: Date,
        updatedAt: Date
    ) {
        self.id = id
        self.userId = userId
        self.title = title
        self.description = description
        self.startAt = startAt
        self.endAt = endAt
        self.allDay = allDay
        self.type = type
        self.location = location
        self.isFixed = isFixed
        self.source = source
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}
